src/encoded/static/components/regulome_search.js

```diff
@@ -544,6 +544,32 @@ const populationOrder = [
     'source unknown',
 ];
 
+// function to de-dup overlapping peaks in each dataset from ChIP, DNase and ATAC-seq assays
+function filter_overlapping_peaks(Datasets) {
+    var Datasets_filtered = [];
+    var Datasets_filtered_count = 0;
+    if (Datasets.length > 0) {
+        //we want to keep the wider peaks in each dataset
+        //if the start & end positions are the same we will keep the one with the strongest signal
+        //sort all peaks by dataset ids -> start positions in ascending order -> end positions in descending order -> signals in descending order
+        Datasets.sort((a,b)=>a.dataset_rel.localeCompare(b.dataset_rel) || a.start-b.start || b.end-a.end || b.value-a.value);
```

Comment:
value is not used for sorting later, do we still need to sort by value?

Reply:
We are showing signals (i.e. values) for ChIP-seq peaks, so I thought it's better to explicitly keep the one with the strongest signal this way (when there are multiple peaks with exactly same positions from the same dataset)?

Reply:
sound good!